api/src/main/java/com/cloud/vm/snapshot/VMSnapshot.java

```diff
@@ -59,6 +59,7 @@ public String getDescription() {
             s_fsm.addTransition(Error, Event.ExpungeRequested, Expunging);
             s_fsm.addTransition(Expunging, Event.ExpungeRequested, Expunging);
             s_fsm.addTransition(Expunging, Event.OperationSucceeded, Removed);
+            s_fsm.addTransition(Expunging, Event.OperationFailed, Ready);
```

Comment:
should the target state be `Error` instead of `Ready` ? maybe the operation half succeeded

Reply:
@DaanHoogland That actually does make sense. 
So, do I then add to the garbage collection thread to clean up those vm snapshots that are in the error state?

Reply:
@Pearl1594 Setting to error state should be fine I think. User or admin can retry deletion of the VM snapshots again. We already have the transition in place. `s_fsm.addTransition(Error, Event.ExpungeRequested, Expunging);`